lib/ddtrace/span.rb

```diff
@@ -80,7 +83,19 @@ def initialize(tracer, name, options = {})
     #
     #   span.set_tag('http.method', request.method)
     def set_tag(key, value = nil)
-      @meta[key] = value.to_s
+      # Keys must be unique between tags and metrics
+      @metrics.delete(key) if @metrics.key?(key)
```

Comment:
Do we need to check for key existence here?
I believe `Hash#delete` is safe on non-existing keys.

Same comment applies to `def set_metric(key, value)` method.